### ISSUE TITLE
Filename extension of some headers changed?

### ISSUE THREAD

**Issue by abrownsword** [skypjack/uvw#195]
In some release since the last one I was using (2.1.0), many (but not all) of the header files changed from being *.hpp to *.h.  This is annoying if you're including specific headers instead of the uvw.hpp.  Why was this done and what is the significance of .h vs .hpp?

**Comment by skypjack** [comment]
With the last versions, we switched to a dual mode library. That is, you can use it as header-only library or as a precompiled one.
Because of that, many class definitions that were contained in their own `.hpp` file (`.h` + `.cpp` file, declaration + definition) have been split in two files.
This is more or less the reason for which many `.hpp` files turned to the `.h` extension.

**Comment by stefanofiorentino** [comment]
I could only confirm what @skypjack said. `uvw` was implemented into .hpp because of the "header-only" status. Once the static/shared came into play, we did split the declarations and definitions in .h/.cpp wherever possible. 
Is your question about compilation time? or what else? 
In case you use static/shared deployment, to include uvw.hpp brings similar effect to include specific headers in header-only configuration.
